contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+        current base classes need extension based on user feedback.
     experiment_type:
       doc: |
         Specify the type of the optical experiment.
-    start_time(NX_DATE_TIME):
+        
+        Chose other if non of these methods are suitable. You may specify
+        fundamental characteristics or properties in the experimental sub-type.
+      enumeration: [ellipsometry, Raman spectroscopy, photoluminescence, transmission spectroscopy, reflection spectroscopy, other]
+    experiment_sub_type:
+      doc: |
+        Specify a special property or characteristic of the experiment, which specifies
+        the generic experiment type.
+      enumeration: [time resolved, imaging, pump-probe, other]
+    # Replace maybe sub_type with exp_technique?
+    #experimental technique --> pump-probe, parameters seperated (i.e. time, space)?
+    # The above listed things are related to beam properties? This is different
+    # compared to temperature or pressure. Is this a useful distinction?
+    experiment_type_other:
       doc: |
-        Start time of the experiment. UTC offset should be specified.
+        If "other" was selected in "experiment_type", specify the experiment here
+        with a free text name, which is knwon in the respective community of interest.
     (NXuser):
+      exists: [min, 0, max, infty]
       doc: |
-        Contact information of at least the user of the instrument or the
-        investigator who performed this experiment.
-        Adding multiple users, if relevant, is recommended.
+        Contact information and eventually details of at persons who
+        performed the measurements. This can be for example the principal
+        investigator or student.
+        It is recommended to add multiple users if relevant.
+
+        Due to data privacy concerns, there is no minimum requirement.
       name(NX_CHAR):
+        exists: recommended
         doc: |
-          Name of the user.
+          Given (first) name and surname of the user.
       affiliation(NX_CHAR):
-        exists: recommended
+        exists: optional
         doc: |
-          Name of the affiliation of the user at the point in time when the
-          experiment was performed.
+          Name of the affiliation of the user at the point in time
+          when the experiment was performed.
       address(NX_CHAR):
-        exists: recommended
+        exists: optional
         doc: |
-          Street address of the user's affiliation.
+          Postal address of the user's affiliation.
       email(NX_CHAR):
+        exists: optional
         doc: |
-          Email address of the user.
+          Email address of the user at the point in time when the experiment
+          was performed. Writing the most permanently used email is recommended.
       orcid(NX_CHAR):
         exists: recommended
         doc: |
           Author ID defined by https://orcid.org/.
       telephone_number(NX_CHAR):
```

Comment:
Is this really something that needs to be part of the app-def? I would never write this in a data file.